scripts/checks/verify_supply_chain.py

```
@@ -645,6 +662,212 @@ def verify_release_asset_allowlist_policy() -> list[str]:
+        for owner, dependency_tokens in package_dependencies.items()
+        if dependency in dependency_tokens
+    }
+
+
+def cargo_lock_has_dependency_chain(
+    package_dependencies: dict[str, list[str]], package_chain: tuple[str, ...]
+) -> bool:
+    """Return whether Cargo dependencies contain the exact package chain."""
+    return all(
+        cargo_dependency_targets_package(package_dependencies, owner, dependency)
+        for owner, dependency in zip(package_chain, package_chain[1:])
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)